src/services/tx/txSender.ts

```diff
@@ -95,358 +94,440 @@ const getRecommendedTxParams = async (
   }
 }
 
-/**
- * Create a transaction from raw params
- */
-export const createTx = async (txParams: SafeTransactionDataPartial, nonce?: number): Promise<SafeTransaction> => {
-  const safeSDK = getAndValidateSafeSDK()
-
-  // If the nonce is not provided, we get the recommended one
-  if (nonce === undefined) {
-    const recParams = (await getRecommendedTxParams(txParams)) || {}
-    txParams = { ...txParams, ...recParams }
-  } else {
-    // Otherwise, we use the provided one
-    txParams = { ...txParams, nonce }
-  }
+export const useTxSender = () => {
```

Comment:
His hook is _really_ verbose. I'm sure there's a more elegant way to wrap all those functions.